tetris.py

```diff
@@ -305,15 +357,21 @@ def move_current_block_down(self):
         """
         Moves the current block downwards one square on the canvas
         """
-        if self.block_hits_bottom_if_it_moves_down():
-            if self.block_letter not in self.landed_blocks:
-                self.landed_blocks[self.block_letter] = []
-            self.landed_blocks[self.block_letter].extend(self.get_block_shape())
-            self.full_line_clear()
-            self.new_block()
-        elif not self.fast_down:
-            x, y = self.current_block_center
-            self.current_block_center = (x, y + 1)
+        if self.game_status == GameStatus.paused:
+            self.main_board.canvas.after(game_speed, self.move_current_block_down)
+        elif self.game_status == GameStatus.in_progress:
+            if self.block_hits_bottom_if_it_moves_down():
+                if self.block_letter not in self.landed_blocks:
+                    self.landed_blocks[self.block_letter] = []
+                self.landed_blocks[self.block_letter].extend(self.get_block_shape())
+                self.full_line_clear()
+                self.game_over()
+                self.new_block()
+            elif not self.fast_down:
+                x, y = self.current_block_center
+                self.current_block_center = (x, y + 1)
+            self.main_board.draw_block(self.get_block_shape(), self.block_letter, self.landed_blocks)
+            self.main_board.canvas.after(game_speed, self.move_current_block_down)
```

Comment:
This can be simplified. Right now it's like:

```
if self.game_status == GameStatus.paused:
    self.main_board.canvas.after(...)
elif self.game_status == GameStatus.in_progress:
    ...lots of code...
    self.main_board.canvas.after(...)
```

This would be simpler:

```
if self.game_status == GameStatus.in_progress:
    ...lots of code...

if self.game_status != GameStatus.game_over:
    self.main_board.canvas.after(...)
```